### PR TITLE
Add SA-Med2D-20M dataset

### PR DESCRIPTION
**Pull request by anwai98**
This is a segmentation dataset for 2d images from various modalities for different tasks.

### CODE REVIEW

**Comment by anwai98** [comment]
@constantinpape Would be good to give this a round of review. Looks good to me, this needs a bit minor checks and refactors from my side, but the core construct is in place. Let me know if you spot something.

PS. I have a few detailed questions, I'll leave comments in those places.

File: torch_em/transform/generic.py [hunk 1]
```diff
@@ -40,7 +40,7 @@ def __init__(self, *transforms):
     def __call__(self, *inputs):
         outputs = self.transforms[0](*inputs)
         for trafo in self.transforms[1:]:
-            outputs = trafo(*outputs)
```
Comment (anwai98) [inline torch_em/transform/generic.py:43]: This was the previous state where the outputs from the first trafo were passed as inputs to the next round of trafos using `*outputs`, however when I used this in my workflow, I noticed that `*outputs` splits the outputs (say an array of shape (3, 512, 512)) to a tuple of three different arrays (of shape (512, 512) each). I removed the `*` and it stopped this behaviour. Could you confirm if this is the way-to-go or am I missing something in the transform functions?

For reference: this happens in "raw_transform", where the first transform is the resize inputs transform, and the next is for the default raw transforms.

Comment (anwai98) [reply torch_em/transform/generic.py:43]: `is_multi_tensor` as another flag to take care of this.

**Review by constantinpape — APPROVED** [review]
This looks good to me now. Anything else to add from your side?

**Comment by anwai98** [comment]
I am waiting for the post-processing to complete and to verify the logic of storing the json file. Once that is done, I will ping here!

**Comment by anwai98** [comment]
Hi @constantinpape,

This is ready and good to go from my side. Thanks!

**Review by constantinpape — APPROVED** [review]

**Pull request merged**